airflow_versions/airflow_versions.go

```diff
@@ -17,9 +27,60 @@ func GetDefaultImageTag(httpClient *Client, airflowVersion string) (string, erro
 		return "", err
 	}
 
+	if httpClient.useAstronomerCertified {
+		return getAstronomerCertifiedTag(resp.AvailableReleases, airflowVersion)
+	}
+
+	return getAstroRuntimeTag(resp.RuntimeVersions, airflowVersion)
+}
+
+// get latest runtime tag associated to provided airflow version
+// if no airflow version is provided, returns the latest astro runtime version available
+func getAstroRuntimeTag(runtimeVersions map[string]RuntimeVersion, airflowVersion string) (string, error) {
+	availableTags := []string{}
+	availableVersions := []string{}
+
+	for runtimeVersion, r := range runtimeVersions {
+		if r.Metadata.Channel != VersionChannelStable {
+			continue
+		}
+
+		// if user wants specific airflow version, get all runtime version associated to this airflow version
+		if r.Metadata.AirflowVersion == airflowVersion {
+			availableTags = append(availableTags, runtimeVersion)
+		} else {
+			availableVersions = append(availableVersions, runtimeVersion)
+		}
+	}
+
+	tagsToUse := availableVersions
+	if airflowVersion != "" {
+		tagsToUse = availableTags
+	}
+
+	if len(tagsToUse) == 0 {
+		return "", ErrNoTagAvailable{airflowVersion: airflowVersion}
+	}
+
+	// get latest runtime version
+	latestVersion, _ := NewAirflowVersion(tagsToUse[0], []string{tagsToUse[0]})
+	for i := 1; i < len(tagsToUse); i++ {
+		tag := tagsToUse[i]
+		nextVersion, err := NewAirflowVersion(tag, []string{tag})
+		if err == nil {
+			if nextVersion.GreaterThan(latestVersion) {
```

Comment:
I went to read the cloud-cli code for this same feature, they look at the runtime version (5.0.0 which is in an `alpha` channel for now and can't be used for production purposes).

Just like I do, I just added another check for the runtime version's release channel to only allow users to use stable releases.

Anyway, we probably won't merge this PR into this repository, since we'll do the convergence soon, and I'll probably just submit a PR to allow AC images and default to AC images if context is software (CLI logged into a gen1 cluster).

But I guess that most of this PR will be dropped since cloud-cli already developped this feature.